src/RahaStellar.ts

```diff
@@ -0,0 +1,97 @@
+
+/**
+ * Note: TransactionMetadata is undocumented and quite ugly to interact with. Mark had to look into the SDK source to do
+ * TODO: Can we use documented features/do this in an easier to understand and less brittle way?
+ */
+function getFirstOperationAttributesFromTransactionMeta(transactionMeta) {
+    return transactionMeta._value[0]._attributes.changes[0]._value._attributes.data._value._attributes
+}
+
+/**
+ * Return the name of a block from transaction metadata.
+ */
+export function getBlockNameFromTransactionMeta(transactionMeta) {
+    return getFirstOperationAttributesFromTransactionMeta(transactionMeta).dataName;
+}
+
+/**
+ * Legacy method for blocks set using only managed data with no memo hash.
+ * See note on TransactionMetadata above.
+ */
+export function getMultiHashFromTransactionMeta(transactionMeta) {
+    const charCode = getFirstOperationAttributesFromTransactionMeta(transactionMeta).dataValue;
+    return String.fromCharCode.apply(null, charCode);
+}
+
+/**
+ * Extract the TransactionMeta XDR object from the Stellar transaction.
+ * See note on TransactionMetadata above.
+ */
+export function getTransactionMetaFromTransaction(stellarTx) {
+    // xdr cast because StellarSdk XDR typing is bad.
+    return (StellarSdk.xdr as any).TransactionMeta.fromXDR(stellarTx.result_meta_xdr, 'base64');
+}
+
+function multiHashToMemo(base58MultiHash) {
+    const bytes = bs58.decode(base58MultiHash);
+    return bytes.slice(2);
+}
+
+class RahaStellar {
+    isTest;
+    horizonServer : StellarSdk.Server;
+
+    constructor (isTest) {
+        this.isTest = isTest;
+        StellarSdk.Network.usePublicNetwork();
+        this.horizonServer = new StellarSdk.Server(STELLAR_ENDPOINT);
+    }
+
+    async getRahaAccount() {
+        return this.horizonServer.loadAccount(RAHA_IO_STELLAR_PUBLIC_KEY);
+    }
+
+    /**
+     * Return all transactions associated with the Raha account from the Stellar blockchain ordered chronologically.
+     * TODO: Documentation doesn't specify the current page limit. Add paging support.
+     * Update: Default limit is 10, max is 200. https://stellar.stackexchange.com/a/792/1187.
+     */
+    async getTransactions() {
+        return (await this.horizonServer.transactions()
+            .forAccount(RAHA_IO_STELLAR_PUBLIC_KEY)
+            .order('desc')
+            .limit(200)
+            .call()).records;
+    }
+
+    /**
+     * Create a new transaction in the Stellar blockchain that records
+     * the multiHash of the new block.
+     */
+    async createRahaBlockchainTransaction(multiHash, secretKey) {
+        // Lol. these types are broken. also, so is the documentation.
+        const memo = new StellarSdk.Memo('hash' as 'MemoHash', multiHashToMemo(multiHash));
+        const transaction = new StellarSdk.TransactionBuilder(await this.getRahaAccount())
+            .addMemo(memo)
+            .build();
+        transaction.sign(StellarSdk.Keypair.fromSecret(secretKey));
```

Comment:
Can we add the manage data operation to give it the name `block-latest`? Then way we can use the same account for multiple forms of meta-data (eg eventually adding FAQ or even code the the blockchain so that it becomes self-describing for how to interpret the data)

Reply:
why do we add the hash as a memo and the name as a data operation? Why not make both data operations?

Reply:
alternately, why not make the transaction type - i.e. `block` in this case, the memo, since there can only be one of those, and the hash part of the data?

Reply:
We don't want to have a bunch of different data keys because that increases reserve needed by 10 XLM/entry (though that [will probably decrease](https://www.reddit.com/r/Stellar/comments/7n4vvt/whats_the_reserve_in_xlm_needed_for_data_storage/)) which could be problematic if we have millions of blocks. Whenever we re-write `block-latest` we lose the old value, it will no longer be available from Horizon servers. 

If we ever need to get around the 32 byte limit, then manage data could be useful for more than just "color coding" the memo hash. Each manage data key can be up to 64 bytes. In a single transaction we could probably write then delete many many <key, data> pairs where data is just dummy data, and a record of the <key> value would still be available afterwards within the transaction. Just interesting side note, but at the moment changing things would have no benefit to members.